config/config.go

```diff
@@ -66,56 +109,85 @@ func ReadConfigFromFile() (ShipperConfig, error) {
 
 // ShipperConfigFromUnitConfig converts the configuration provided by Agent to the internal
 // configuration object used by the shipper.
-// Currently this just converts the given struct to json and tries to deserialize it into the
-// ShipperConfig struct. This is not the right way to do this, but this gets the build and
-// tests passing again with the new version of elastic-agent-client. Migrating fully to this
-// new config structure is part of the overall agent V2 transition, for more details see
-// https://github.com/elastic/elastic-agent/issues/617.
-func ShipperConfigFromUnitConfig(logLevel client.UnitLogLevel, config *proto.UnitExpectedConfig) (ShipperConfig, error) {
-	jsonConfig, err := config.GetSource().MarshalJSON()
-	if err != nil {
-		return ShipperConfig{}, err
+func ShipperConfigFromUnitConfig(level client.UnitLogLevel, rawConfig *proto.UnitExpectedConfig) (ShipperRootConfig, error) {
+	cfgObject := DefaultConfig()
+
+	// set config based on the lowest log level
+	lowestLevel := client.UnitLogLevelError
+	if level > lowestLevel {
+		logp.L().Debugf("Got new log level: %s", level.String())
```

Comment:
You should be looking to see if the level was increased or decreased vs the current log level. The Beats implementation doesn't need to do this because it always has to reload the entire Beat configuration on any unit change because of the way the object reloader does configuration diffs. That restriction doesn't apply to the shipper.

Reply:
hmm, yah, this logic is weird.